pyluba/bluetooth/ble_message.py

```diff
@@ -1052,4 +1054,210 @@ def _getPackageType(self, typeValue: int):
+
+    def wifi_connectinfo_update2(self):
+        hash_map = {"getMsgCmd": 1}
+        self.post_custom_data(self.get_json_string(68, hash_map))
+
+    def get_record_wifi_list(self, is_binary):
+        print(f"getRecordWifiList().isBinary={is_binary}")
+        if is_binary:
+            build = dev_net_pb2.DevNet(
+                todev_ble_sync=1, todev_wifi_list_upload=dev_net_pb2.DrvWifiList())
+            print("Send command - get memorized WiFi list upload command")
+            self.send_order_msg_net(build, 6, True)
+            return
+        self.get_record_wifi_list2()
+
+    def get_record_wifi_list2(self):
+        self.post_custom_data(self.get_json_string(69))
+
+    def send_blue_tooth_device_sync(self, link_type, device_name_temp, log_type):
+        build = dev_net_pb2.DevNet(todev_ble_sync=link_type)
+        print(
+            f"Send Bluetooth heartbeat======sendBleSync=======:{link_type}  connectDeviceName:{device_name_temp}  logType:{log_type}")
+        print("Send heartbeat print---send Bluetooth heartbeat")
+        self.send_order_msg_net(build, 7, False)
+
+    def get_device_version_main(self, device_name):
+        new_builder = dev_net_pb2.DrvDevInfoReq.Builder()
+        new_builder.add_req_ids(dev_net_pb2.DrvDevInfoReqId.Builder().set_id(1).set_type(6).build())
+        build = dev_net_pb2.DevNet.Builder().set_todev_devinfo_req(new_builder.build()).build()
+        print("Send command -- Get main version number")
+        print("Send command -- Get main version number:")
+        self.send_order_msg_net(build, 41, True)
+
+    def get_device_version_main2(self):
+        new_builder = dev_net_pb2.DrvDevInfoReq.Builder()
+        new_builder.add_req_ids(dev_net_pb2.DrvDevInfoReqId.Builder().set_id(1).set_type(6).build())
+        build = dev_net_pb2.DevNet.Builder().set_todev_devinfo_req(new_builder.build()).build()
+        print("Send command -- Get main version number")
+        print("Send command -- Get main version number:")
+        self.send_order_msg_net(build, 41, True)
+        
+    def close_clear_connect_current_wifi(self, ssid, status, is_binary):
+        if is_binary:
+            build = dev_net_pb2.DevNet(
+                todev_ble_sync=1,
+                todev_wifi_configuration=dev_net_pb2.DrvWifiSet(
+                    config_param=status,
+                    confssid=ssid
+                )
+            )
+            print(f"Send command - set network (disconnect, direct connect, forget, no operation reconnect) operation command (downlink ssid={ssid}, status={status})")
+            self.send_order_msg_net(build, 7, True)
+            return
+        self.close_clear_connect_current_wifi2(ssid, status)
+
+    def close_clear_connect_current_wifi2(self, ssid, get_msg_cmd):
+        data = {
+            "ssid": ssid,
+            "getMsgCmd": get_msg_cmd
+        }
+        self.send_raw_data_callback.send_data(self.get_json_string(77, data).encode())
```

Comment:
This method exists under a different name, should see other examples in this file.